source/wrapper/contracts/Wrapper.sol

```diff
@@ -74,42 +75,65 @@ contract Wrapper {
     require(order.signature.v != 0,
       "SIGNATURE_MUST_BE_SENT");
 
-    // The sender is sending ether that must be wrapped.
-    if (order.sender.token == address(wethContract)) {
+    // Check if sender is sending ether that must be wrapped.
+    _wrapEther(order.sender);
 
-      // Ensure message value is sender param.
-      require(order.sender.param == msg.value,
+    // Perform the swap.
+    swapContract.swap(order);
+
+    // Check if sender is receiving ether that must be unwrapped.
+    _unwrapEther(order.sender.wallet, order.signer.token, order.signer.param);
```

Comment:
// Perform unwrapping weth action when sender receives weth, no action is taken otherwise.

update the comment as the function performs a state change